WIP: Add additional detail to appdata files

pwithnall opened the pull request:
In particular, add a `<bundle/>` tag which allows the appdata to be
linked to a flatpak.

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T29392

pwithnall commented:
This is WIP and should not be merged, because I’m not sure how best to integrate the substitution of `SDK_BRANCH`, `BRANCH` and `SDK_ARCH` into the build system in this module. Should I rename the `.xml` files to `.xml.in` and add some code to `generate-manifests` which outputs the `.xml` with variables substituted for the current `args.arch`?

dbnicholson commented on com.endlessm.BaseEknServicesMultiplexer.metainfo.xml:
```diff
@@ -7,4 +7,6 @@
   <summary>Extension points for Endless Desktop Search</summary>
   <url type="homepage">https://endlessos.com/</url>
+  <project_group>Endless</project_group>
+  <bundle type="flatpak" runtime="com.endlessm.apps.Platform/@SDK_ARCH@/@SDK_BRANCH@" sdk="com.endlessm.apps.Sdk/@SDK_ARCH@/@SDK_BRANCH@">runtime/com.endlessm.EknServicesMultiplexer/@SDK_ARCH@/@BRANCH@</bundle>
```
I think this should be BaseEknServicesMultiplexer, right?

pwithnall replied:
I think Damián fixed this.

dnohales commented:
I've updated the `generate-manifests` script to do all the replacements and modifications in just one step.

The `eos-build` part of this is here: https://github.com/endlessm/eos-build/pull/1126

Let me know what you think since I'm not really familiar with this project.

dbnicholson commented on generate-manifests:
```diff
@@ -3,28 +3,54 @@
 from argparse import ArgumentParser
 import json
 import os
+import re
```
I don't think you ended up using `re` anywhere since you're just doing a `str.replace`.

pwithnall replied:
Dropped it.

dbnicholson commented:
bot, retest this please

pwithnall commented:
I addressed the nitpick, fixed a few pyflakes/pylint warnings, and split the one commit in two — one with Damián’s work to refactor `generate-manifests`, and one with my appdata additions.
